File: src/main/java/com/github/njuro/jard/ban/BanRepository.java
package com.github.njuro.jard.ban;

import com.github.njuro.jard.base.BaseRepository;
import com.github.njuro.jard.user.User;
import java.time.OffsetDateTime;
import java.util.List;
import java.util.Optional;
import org.springframework.stereotype.Repository;

@Repository
public interface BanRepository extends BaseRepository<Ban> {

  List<Ban> findByIp(String ip);

  List<Ban> findByStatusAndValidToBefore(BanStatus status, OffsetDateTime dateTime);

  Optional<Ban> findByIpAndStatus(String ip, BanStatus status);

  List<Ban> findByBannedBy(User bannedBy);
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/ban/BanRepository.kt
package com.github.njuro.jard.rewrite.ban

import com.github.njuro.jard.rewrite.base.BaseRepository
import com.github.njuro.jard.rewrite.user.User
import org.springframework.stereotype.Repository
import java.time.OffsetDateTime

@Repository
interface BanRepository : BaseRepository<Ban> {
    fun findByIp(ip: String): List<Ban>
    fun findByStatusAndValidToBefore(status: BanStatus, dateTime: OffsetDateTime): List<Ban>
    fun findByIpAndStatus(ip: String, status: BanStatus): Ban?
    fun findByBannedBy(bannedBy: User): List<Ban>
}
